labs/b3.md

```diff
@@ -0,0 +1,644 @@
+modern interpreted languages. For this reason, more complex scripts are 
+popularly written in another, more general, programming language like `python`. 
+Scripting with `python` is increasingly popular among sysadmins.
+
+Countless great [tutorials][7] for learning `python` are available online. 
+Alternatively, Berkeley offers the self-paced course [CS 9H][7] to students 
+with a programming background, or [CS 61A][8] as a `python`-based introductory 
+course in computer science.
+
+Adapting `python` to command-line scripting is only a matter of using relevant 
+modules. Here are some tips:
+
+- The [`argparse`][9] module in the `python` standard library is a popular way 
+to implement command-line interfaces for `python` scripts
+- [`fabric`][10] simplifies some sysadmin tasks, mostly in regard to 
+application deployment
+- [`salt`][11] is useful for general infrastructure management
+- [`psutil`][12] provides an interface to system information monitoring
+
+In practice, the decision to write a script in either `python` or `bash` is 
+largely dependent on the context of the task at hand. Generally, tasks solvable 
+with simple shell commands and those requiring simple file reading, writing, 
+and appending are often a good fit for a `bash` script. Those with complex 
+control logic, recursion, and other more general programming patterns are a 
+better fit for a `python` script.
+
+--------------------------------------------------------------------------------
+
+# Scripting Lab Assignment
+
+You'll be completing a classic first shell scripting assignment: make a 
+phonebook.
+
+Write a shell script `phonebook` which has the following behavior:
+
+- `./phonebook new name number` adds an entry to the phonebook. and
+- `./phonebook name` displays the name and phone number associated with that 
+name.
+
+- `./phonebook list` displays every entry in the phonebook (in no particular 
+order). If the phonebook has no entries, display `phonebook is empty`
+
+- `./phonebook remove name` deletes the entry associated with that name.
+
+- `./phonebook clear` deletes the entire phonebook.
+
+For example,
+
+```
+$ ./phonebook new Linus Torvalds 101-110-0111
+$ ./phonebook Linus Torvalds
+Linus Torvalds 101-110-1010
+$ ./phonebook new Tux Penguin 555-666-7777
+$ ./phonebook list
+Linus Torvalds 101-110-1010
+Tux Penguin 555-666-7777
+$ ./phonebook remove Linus Torvalds
+$ ./phonebook list
+Tux Penguin 555-666-7777
+$ ./phonebook clear
+$ ./phonebook list
+phonebook is empty
+```
+
+Here's the kicker: **You have to implement this same functionality in both 
+`bash` and `python`.** This is to help illuminate the strengths and weaknesses 
+of each language in the context of writing a simple CLI application.
+
+To help you in this task, skeleton code for this lab can be found 
+[here](https://github.com/0xcf/decal-labs/tree/master/b3).
+Once you are done with this task, the checkoff form can be found 
+[here](https://goo.gl/forms/mWXCVzZxhL0ytIdG2).
```

Comment:
This is the only thing that I have to change, since I want to put this on Gradescope. I think I will try to set this up with some basic tests so that we can have an autograder that handles the grading.